regression-test/data/nereids_function_p0/scalar_function/Q.out

```diff
@@ -86,3 +86,135 @@
 1
 1
 
+-- !sql_quote_varchar_notenull --
+'varchar11'
+'varchar11'
+'varchar11'
+'varchar11'
+'varchar12'
+'varchar12'
+'varchar12'
+'varchar12'
+'varchar13'
+'varchar13'
+'varchar13'
+'varchar13'
+
+-- !sql_quote_varchar_notenull --
+'LINESTRING (1 1, 2 2)'
+'LINESTRING (1 1, 2 2)'
+'LINESTRING (1 1, 2 2)'
+'LINESTRING (1 1, 2 2)'
+'LINESTRING (1 1, 2 2)'
+'LINESTRING (1 1, 2 2)'
+'LINESTRING (1 1, 2 2)'
+'LINESTRING (1 1, 2 2)'
+'LINESTRING (1 1, 2 2)'
+'LINESTRING (1 1, 2 2)'
+'LINESTRING (1 1, 2 2)'
+'LINESTRING (1 1, 2 2)'
+
+-- !sql_quote_varchar_notenull --
+'POLYGON ((0 0, 10 0, 10 10, 0 10, 0 0))'
+'POLYGON ((0 0, 12 0, 10 16, 1 10, 0 0))'
+'POLYGON ((0 0, 38 4, 38 37, 4 1, 0 0))'
+'POLYGON ((1 1, 11 4, 42 44, 4 18, 1 1))'
+'POLYGON ((1 1, 16 1, 16 16, 1 16, 1 1))'
+'POLYGON ((1 1, 34 1, 34 34, 1 34, 1 1))'
+'POLYGON ((1 1, 4 1, 4 4, 1 4, 1 1))'
+'POLYGON ((1 1, 5 0, 10 4, 4 3, 1 1))'
+'POLYGON ((1 1, 56 0, 67 89, 4 32, 1 1))'
+'POLYGON ((1 1, 8 0, 48 34, 4 10, 1 1))'
+'POLYGON ((1 1, 9 4, 10 4, 4 1, 1 1))'
+'POLYGON ((4 4, 45 4, 45 45, 4 45, 4 4))'
+
+-- !sql_quote_varchar --
+'null'
+'varchar11'
+'varchar11'
+'varchar11'
+'varchar11'
+'varchar12'
+'varchar12'
+'varchar12'
+'varchar12'
+'varchar13'
+'varchar13'
+'varchar13'
+'varchar13'
+
+-- !sql_quote_string --
+'null'
+'string1'
+'string1'
+'string1'
+'string1'
+'string2'
+'string2'
+'string2'
+'string2'
+'string3'
+'string3'
+'string3'
+'string3'
+
+-- !sql_quote_string --
+'LINESTRING (1 1, 2 2)'
+'LINESTRING (1 1, 2 2)'
+'LINESTRING (1 1, 2 2)'
+'LINESTRING (1 1, 2 2)'
+'LINESTRING (1 1, 2 2)'
+'LINESTRING (1 1, 2 2)'
+'LINESTRING (1 1, 2 2)'
+'LINESTRING (1 1, 2 2)'
+'LINESTRING (1 1, 2 2)'
+'LINESTRING (1 1, 2 2)'
+'LINESTRING (1 1, 2 2)'
+'LINESTRING (1 1, 2 2)'
+'null'
+
+-- !sql_quote_string --
+'POLYGON ((0 0, 10 0, 10 10, 0 10, 0 0))'
+'POLYGON ((0 0, 12 0, 10 16, 1 10, 0 0))'
+'POLYGON ((0 0, 38 4, 38 37, 4 1, 0 0))'
+'POLYGON ((1 1, 11 4, 42 44, 4 18, 1 1))'
+'POLYGON ((1 1, 16 1, 16 16, 1 16, 1 1))'
+'POLYGON ((1 1, 34 1, 34 34, 1 34, 1 1))'
+'POLYGON ((1 1, 4 1, 4 4, 1 4, 1 1))'
+'POLYGON ((1 1, 5 0, 10 4, 4 3, 1 1))'
+'POLYGON ((1 1, 56 0, 67 89, 4 32, 1 1))'
+'POLYGON ((1 1, 8 0, 48 34, 4 10, 1 1))'
+'POLYGON ((1 1, 9 4, 10 4, 4 1, 1 1))'
+'POLYGON ((4 4, 45 4, 45 45, 4 45, 4 4))'
+'null'
+
+-- !sql_quote_string --
+'ST_Point(11.1, 31.1)'
+'ST_Point(12.1, 32.1)'
+'ST_Point(12.1, 32.1)'
+'ST_Point(12.2, 32.2)'
+'ST_Point(12.3, 32.3)'
+'ST_Point(12.4, 32.4)'
+'ST_Point(12.5, 32.5)'
+'ST_Point(12.6, 32.6)'
+'ST_Point(12.7, 32.7)'
+'ST_Point(17.1, 37.1)'
+'ST_Point(19.1, 39.1)'
+'ST_Point(22.1, 22.1)'
+'null'
+
+-- !sql_quote_string --
+'ST_Point(11.1, 31.1)'
+'ST_Point(12.1, 32.1)'
+'ST_Point(12.1, 32.1)'
+'ST_Point(12.2, 32.2)'
+'ST_Point(12.3, 32.3)'
+'ST_Point(12.4, 32.4)'
+'ST_Point(12.5, 32.5)'
+'ST_Point(12.6, 32.6)'
+'ST_Point(12.7, 32.7)'
+'ST_Point(17.1, 37.1)'
+'ST_Point(19.1, 39.1)'
+'ST_Point(22.1, 22.1)'
+'null'
```

Comment:
what's the input row? if "null" string literal, it's right. but if null value, should be null. the function shouldn't deal it.